modules/hostap/src/supp_events.c

```diff
@@ -9,10 +9,134 @@
 #include "includes.h"
 #include "common.h"
 
-int send_wifi_mgmt_event(const char *ifname, enum net_event_wifi_cmd event, int status)
+#define MAC_STR_FORMAT "%2hhx:%2hhx:%2hhx:%2hhx:%2hhx:%2hhx"
+
+static char *wpa_supp_event_map[] = {
+	"CTRL-EVENT-CONNECTED",
+	"CTRL-EVENT-DISCONNECTED",
+	"CTRL-EVENT-ASSOC-REJECT",
+	"CTRL-EVENT-AUTH-REJECT",
+	"CTRL-EVENT-TERMINATING",
+	"CTRL-EVENT-SSID-TEMP-DISABLED",
+	"CTRL-EVENT-SSID-REENABLED",
+	"CTRL-EVENT-SCAN-STARTED",
+	"CTRL-EVENT-SCAN-RESULTS",
+	"CTRL-EVENT-SCAN-FAILED",
+	"CTRL-EVENT-BSS-ADDED",
+	"CTRL-EVENT-BSS-REMOVED",
+	"CTRL-EVENT-NETWORK-NOT-FOUND",
+	"CTRL-EVENT-NETWORK-ADDED",
+	"CTRL-EVENT-NETWORK-REMOVED",
+};
+
+static int process_status(struct supp_int_event_data *event_data, char *status)
+{
+	int ret = 1; /* For cases where parsing is not being done*/
+	int event = -1;
+	int i;
+	unsigned char *mac;
+	union supp_event_data *data;
+
+	data = (union supp_event_data *)event_data->data;
+
+	for (i = 0; i < ARRAY_SIZE(wpa_supp_event_map); i++) {
+		if (strncmp(status, wpa_supp_event_map[i], strlen(wpa_supp_event_map[i])) == 0) {
+			event = i;
+			break;
+		}
+	}
+
+	if (i >= ARRAY_SIZE(wpa_supp_event_map)) {
+		wpa_printf(MSG_ERROR, "Event not supported\n");
```

Comment:
better to log the event as well for debug.

Reply:
Done